### PR TITLE
Ensure that `id->idReg2()` is properly encoded for 4-Byte SIMD instructions

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves https://github.com/dotnet/coreclr/issues/16769

### CODE REVIEW

**Comment by tannergooding** [comment]
FYI. @fiigii, @CarolEidt

**Comment by tannergooding** [comment]
test Windows_NT x64 Checked jitincompletehwintrinsic
test Windows_NT x64 Checked jitx86hwintrinsicnoavx
test Windows_NT x64 Checked jitx86hwintrinsicnoavx2
test Windows_NT x64 Checked jitx86hwintrinsicnosimd
test Windows_NT x64 Checked jitnox86hwintrinsic

test Windows_NT x86 Checked jitincompletehwintrinsic
test Windows_NT x86 Checked jitx86hwintrinsicnoavx
test Windows_NT x86 Checked jitx86hwintrinsicnoavx2
test Windows_NT x86 Checked jitx86hwintrinsicnosimd
test Windows_NT x86 Checked jitnox86hwintrinsic

test Ubuntu x64 Checked jitincompletehwintrinsic
test Ubuntu x64 Checked jitx86hwintrinsicnoavx
test Ubuntu x64 Checked jitx86hwintrinsicnoavx2
test Ubuntu x64 Checked jitx86hwintrinsicnosimd
test Ubuntu x64 Checked jitnox86hwintrinsic

**Comment by tannergooding** [comment]
test Windows_NT x64 Checked jitstressregs0x1000
test Windows_NT x64 Checked jitstressregs0x10
test Windows_NT x64 Checked jitstressregs0x80
test Windows_NT x64 Checked jitstressregs1
test Windows_NT x64 Checked jitstressregs2
test Windows_NT x64 Checked jitstressregs3
test Windows_NT x64 Checked jitstressregs4
test Windows_NT x64 Checked jitstressregs8

test Windows_NT x86 Checked jitstressregs0x1000
test Windows_NT x86 Checked jitstressregs0x10
test Windows_NT x86 Checked jitstressregs0x80
test Windows_NT x86 Checked jitstressregs1
test Windows_NT x86 Checked jitstressregs2
test Windows_NT x86 Checked jitstressregs3
test Windows_NT x86 Checked jitstressregs4
test Windows_NT x86 Checked jitstressregs8

test Ubuntu x64 Checked jitstressregs0x1000
test Ubuntu x64 Checked jitstressregs0x10
test Ubuntu x64 Checked jitstressregs0x80
test Ubuntu x64 Checked jitstressregs1
test Ubuntu x64 Checked jitstressregs2
test Ubuntu x64 Checked jitstressregs3
test Ubuntu x64 Checked jitstressregs4
test Ubuntu x64 Checked jitstressregs8

File: src/jit/emitxarch.cpp [hunk 1]
```diff
@@ -12551,6 +12545,8 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
             assert(IsAVXInstruction(ins));
 
             code = insCodeRM(ins);
+            code = AddVexPrefixIfNeeded(ins, code, size);
+            code = insEncodeReg3456(ins, id->idReg2(), size, code); // encode source operand reg in 'vvvv' bits in 1's complement form
```
Comment (fiigii) [inline src/jit/emitxarch.cpp]: So we unconditionally encode reg2 in VEX.vvvv field and special guys (may need to encode other operand in VEX.vvvv) will be handled in codegen or other places. Right?

Comment (tannergooding) [reply src/jit/emitxarch.cpp]: Yes, that is what we were already doing (for one of the code paths).

If the instruction has special encoding requirements, we should probably assert that they don't call `insEncodeReg3456` and are instead handled elsewhere.

Do you have a list of which instructions meet this requirement?

**Comment by tannergooding** [comment]
Applied formatting patch.

**Comment by tannergooding** [comment]
test Windows_NT x64 Checked jitincompletehwintrinsic
test Windows_NT x64 Checked jitx86hwintrinsicnoavx
test Windows_NT x64 Checked jitx86hwintrinsicnoavx2
test Windows_NT x64 Checked jitx86hwintrinsicnosimd
test Windows_NT x64 Checked jitnox86hwintrinsic

test Windows_NT x86 Checked jitincompletehwintrinsic
test Windows_NT x86 Checked jitx86hwintrinsicnoavx
test Windows_NT x86 Checked jitx86hwintrinsicnoavx2
test Windows_NT x86 Checked jitx86hwintrinsicnosimd
test Windows_NT x86 Checked jitnox86hwintrinsic

test Ubuntu x64 Checked jitincompletehwintrinsic
test Ubuntu x64 Checked jitx86hwintrinsicnoavx
test Ubuntu x64 Checked jitx86hwintrinsicnoavx2
test Ubuntu x64 Checked jitx86hwintrinsicnosimd
test Ubuntu x64 Checked jitnox86hwintrinsic

**Comment by tannergooding** [comment]
test Windows_NT x64 Checked jitstressregs0x1000
test Windows_NT x64 Checked jitstressregs0x10
test Windows_NT x64 Checked jitstressregs0x80
test Windows_NT x64 Checked jitstressregs1
test Windows_NT x64 Checked jitstressregs2
test Windows_NT x64 Checked jitstressregs3
test Windows_NT x64 Checked jitstressregs4
test Windows_NT x64 Checked jitstressregs8

test Windows_NT x86 Checked jitstressregs0x1000
test Windows_NT x86 Checked jitstressregs0x10
test Windows_NT x86 Checked jitstressregs0x80
test Windows_NT x86 Checked jitstressregs1
test Windows_NT x86 Checked jitstressregs2
test Windows_NT x86 Checked jitstressregs3
test Windows_NT x86 Checked jitstressregs4
test Windows_NT x86 Checked jitstressregs8

test Ubuntu x64 Checked jitstressregs0x1000
test Ubuntu x64 Checked jitstressregs0x10
test Ubuntu x64 Checked jitstressregs0x80
test Ubuntu x64 Checked jitstressregs1
test Ubuntu x64 Checked jitstressregs2
test Ubuntu x64 Checked jitstressregs3
test Ubuntu x64 Checked jitstressregs4
test Ubuntu x64 Checked jitstressregs8

**Comment by tannergooding** [comment]
@fiigii, @CarolEidt, @4creators. This brings all of our jitstressregs tests back to passing.

**Comment by CarolEidt** [comment]
LGTM

@fiigii said in #14523:
> Encoding VEX.NDS can get rid of IsDstSrcSrcAVXInstruction.

I assume that would eliminate the need to "unconditionally encode reg2 in VEX.vvvv field and special guys (may need to encode other operand in VEX.vvvv) will be handled in codegen or other places."

For now this seems reasonable.

**Pull request merged**